substratevm/mx.substratevm/testhello.py

```diff
@@ -0,0 +1,227 @@
+    exec_string = execute("backtrace")
+    checker = Checker("backtrace hello.Hello::main",
+                      ["#0[ \t]+hello\\.Hello::main\\(java\\.lang\\.String\\[\\]\\).* at hello/Hello\\.java:64",
+                       "#1[ \t]+0x[0-9a-f]+ in com\\.oracle\\.svm\\.core\\.code\\.IsolateEnterStub::JavaMainWrapper_run_.* at [a-z/]+/JavaMainWrapper\\.java:[0-9]+"
+                      ])
+    checker.check(exec_string, skip_fails=False)
+
+    # look up PrintStream::println methods
+    # expect "All functions matching regular expression "java.io.PrintStream::println":"
+    # expect ""
+    # expect "File java.base/java/io/PrintStream.java:"
+    # expect "      void java.io.PrintStream::println(java.lang.Object)(void);"
+    # expect "      void java.io.PrintStream::println(java.lang.String)(void);"
+    exec_string = execute("info func java.io.PrintStream::println")
+#    checker = Checker("info func java.io.PrintStream::println",
+#                      ["All functions matching regular expression \"java\\.io\\.PrintStream::println\":",
+#                       "",
+#                       "File .*java/io/PrintStream.java:",
+#                       "[ \t]*void java.io.PrintStream::println\\(java\\.lang\\.Object\\)\\(void\\);",
+#                       "[ \t]*void java.io.PrintStream::println\\(java\\.lang\\.String\\)\\(void\\);",
+#                      ])
+    checker = Checker("info func java.io.PrintStream::println",
+                      "[ \t]*void java.io.PrintStream::println\\(java\\.lang\\.String\\)")
+    checker.check(exec_string)
+
+    # set a break point at PrintStream::println(String)
+    # expect "Breakpoint 2 at 0x[0-9a-f]+: java.base/java/io/PrintStream.java, line [0-9]+."
+    exec_string = execute("break java.io.PrintStream::println(java.lang.String)")
+    rexp = "Breakpoint 2 at 0x([0-9a-f]+): file .*java/io/PrintStream\\.java, line [0-9]+\\."
+    checker = Checker('break println', rexp)
+    checker.check(exec_string, skip_fails=False)
+
+    # step into method call
+    execute("step")
+
+    # list current line
+    # expect "31	            if (args.length == 0) {"
+    exec_string = execute("list")
+    rexp = "31[ \t]+if \\(args\\.length == 0\\) {"
+    checker = Checker('list hello.Hello.Greeter::greeter', rexp)
+    checker.check(exec_string, skip_fails=False)
+
+    # run a backtrace
+    # expect "#0  hello.Hello.greeter::greeter(java.lang.String[]).* at hello.Hello.java:31"
+    # expect "#1  0x[0-9a-f]+ in hello.Hello::main(java.lang.String[]).* at hello.Hello.java:64"
+    # expect "#2  0x[0-9a-f]+ in com.oracle.svm.core.code.IsolateEnterStub::JavaMainWrapper_run_.* at [a-z/]+/JavaMainWrapper.java:[0-9]+"
+    exec_string = execute("backtrace")
+    checker = Checker("backtrace hello.Hello.Greeter::greeter",
+                      ["#0[ \t]+hello\\.Hello\\.Greeter::greeter\\(java\\.lang\\.String\\[\\]\\).* at hello/Hello\\.java:31",
```

Comment:
Using raw strings will make you patterns much more readable.
https://docs.python.org/3/library/re.html?highlight=raw%20string%20notation

Reply:
done